src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+				final HashMap<String, String> attrMapPlayer = getAttributesMap(playerListChildNode.getAttributes());
+				s_playerAlliance.put(attrMapPlayer.get("player"), attrMapPlayer.get("alliance"));
+			}
+		}
+	}
+
+	private static void parseMapNode(NodeList mapChildNodes) {
+		for (int map_i = 0; map_i < mapChildNodes.getLength(); ++map_i) {
+			final Node mapChildNode = mapChildNodes.item(map_i);
+			final String mapChildNodeName = mapChildNode.getNodeName();
+			if (mapChildNodeName.equals("territory")) {
+				final NamedNodeMap terrAttrNodes = mapChildNode.getAttributes();
+				String terrName = null;
+				final HashMap<DEFINITION, Boolean> terrDef = new HashMap<TerritoryDefinitionDialog.DEFINITION, Boolean>();
+				for (int terrAttr_i = 0; terrAttr_i < terrAttrNodes.getLength(); ++terrAttr_i) {
+					final Node terrAttrNode = terrAttrNodes.item(terrAttr_i);
+					if (terrAttrNode.getNodeName().equals("name")) {
+						terrName = terrAttrNode.getNodeValue();
+					} else {
+						terrDef.put(TerritoryDefinitionDialog.valueOf(terrAttrNode.getNodeName()), Boolean.valueOf(terrAttrNode.getNodeValue()));
+					}
+				}
+				s_territoyDefintions.put(terrName, terrDef);
+			} else if (mapChildNodeName.equals("connection")) {
+				final NamedNodeMap connectionAttrNodes = mapChildNode.getAttributes();
+				String t1Name = connectionAttrNodes.item(0).getNodeValue();
+				String t2Name = connectionAttrNodes.item(1).getNodeValue();
+				if (t1Name.compareTo(t2Name) > 0) {
+					final String swapHelper = t1Name;
+					t1Name = t2Name;
+					t2Name = swapHelper;
+				}
+				LinkedHashSet<String> t1Connections = s_territoyConnections.get(t1Name);
+				if (t1Connections != null)
+					t1Connections.add(t2Name);
+				else {
+					t1Connections = new LinkedHashSet<String>();
+					t1Connections.add(t2Name);
+					s_territoyConnections.put(t1Name, t1Connections);
+				}
+			}
+		}
+	}
+
+	static void saveXML() {
+		try {
+			final String fileName = new FileSave("Where to Save the Game XML ?", s_xmlStrings.get("info_@name") + ".xml", MapXMLCreator.s_mapFolderLocation).getPathString();
+			if (fileName == null) {
+				return;
+			}
+
+			// write the content into xml file
+			TransformerFactory transformerFactory = TransformerFactory.newInstance();
+			// transformerFactory.setAttribute("indent-number", new Integer(2));
+			Transformer transformer = transformerFactory.newTransformer();
+			transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+			transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+			transformer.setOutputProperty(OutputKeys.DOCTYPE_SYSTEM, "game.dtd");
+			DOMSource source = new DOMSource(getXMLDocument());
+			final File newFile = new File(fileName);
+			StreamResult result = new StreamResult(newFile);
+
+			// Output to console for testing
+			// StreamResult result = new StreamResult(System.out);
+
+			transformer.transform(source, result);
+
+			System.out.println("");

```

Comment:
`ClientLogger.logQuietly( "");` Using that API Ii'll go to the error console and won't pop-up the error console. 

But, it's usage is not appropriate here. 

Event logging is one thing, another to log errors, but trace logging can be an issue.
- it's useful while debugging, but typically only for the one developer who is writing code, often your debug statements are not of value or interest if I were to debug something else nearby in the code.
- The debug statements are coupled to your code, this is more things to update when you change things. It's small, but picture changing a variable:
  - update all comments that mention the variable
  - update any debug statements
  - update any variable names, since now "fifth" might be "sixth", and so on.
Those last 3 have real cost, so they need to really add value to be justified.

Really the problem is that the overall class/method is complex. The answer is not trace logging but to abstract/encapsulate/hide the complexity+logic into helper functions and modules. Then ideally add automated tests to those modules. Then you can trace through the modules more easily, and you'll often find they are simple/small enough that it's easy to trace through what is happening.